### ISSUE TITLE
Underscores don't show up in thought/speech bubbles on Windows

### ISSUE THREAD

**Issue by brollb** [jmoenig/Snap--Build-Your-Own-Blocks#945]
Underscores don't show up in sprite thought or speech bubbles on Windows (they basically appear as spaces).

**Comment by jmoenig** [comment]
Hmm. I don't have Windows myself. Can anybody else confirm this? Is this a new behavior? Which browser / Windows version / device pixel ratio? Thanks!

**Comment by brollb** [comment]
I tried it on 3 Windows computers (2 of them running Windows 7 and one Windows 10) using IE, Edge, Firefox and Chrome. I am not sure about the device pixel ratio but I can look into it.

**Comment by jmoenig** [comment]
is it something that just started today (in other words, do you think my update from earlier today caused it)?

**Comment by cycomachead** [comment]
This is an ollllld issue. It's because I different font is being used, let me search, IIRC this first appeared 2 years ago.

**Comment by cycomachead** [comment]
#503 is the other issue, but I do agree with Nathan's analysis.

@brollb are lowercase "g"s also cutoff?

**Comment by cycomachead** [comment]
Also, could you post a screenshot of the Snap<i>!</i> interface showing this? (To see if the display fonts are somehow different...)

Do you have Chrome on your PC?

**Comment by cycomachead** [comment]
@jmoenig Even on Safari I'm nothing that the botton of the lower case g is cut off. (15" rMBP, Safari 9)

<img width="333" alt="screen shot 2015-09-23 at 3 38 32 pm" src="https://cloud.githubusercontent.com/assets/1505907/10060886/401d4aba-6209-11e5-9d72-0d7775e0dd01.png">

**Comment by jmoenig** [comment]
yep, I can reproduce this, too. Strange, I swear this used to look alright before. So, I just compared the latest version with older ones - alas on current browser versions - and the issue is *not* due to yesterday's rendering precision change (which is always my first guess when somebody discovers something).

**Comment by jmoenig** [comment]
So, one interesting observation is that all the characters appear to show up correctly (for me) in the input slots, but not in the speech bubble. @cycomachead and @brollb do you see the same?

**Comment by jmoenig** [comment]
Oh, wait! I think I found something...

**Comment by cycomachead** [comment]
@jmoenig Yes and no... They are not cut off in input slots (g's or j's) but there is no padding like there is above H or a ` .

**Comment by jmoenig** [comment]
... this smells like an off-by-one glitch. I'll push a fix shortly, then I'd be interested to get your feedback whether that takes care of it.

**Comment by jmoenig** [comment]
Okay, I've got a solution to this and it does work, but I'm not happy with it. The underlying issue is that a few years ago I had to round more floats in canvas coordinates than I had wanted to because some browsers were more picky than others (especially Firefox and IE), and this is causing more *slight* display inaccuracies elsewhere, but nothing that affects the functionality of Snap. I've already started carefully taking out some of those display rounding operations, and I'm going to continue with this, and eventually this will fix a bunch of those "slight" display glitches that occasionally happen. But I'd rather do it by *taking out* the kludges I know are causing this than by introducing more kludges that work around the old ones (the solution I have right now is to shift text rendering up by one pixel, and then to fix the blocks layout accordingly. This works, but it's making things worse and harder to debug than revisiting the old workarounds). 

I've just now tested some of those, and unfortunately taking out more of those old rounding workarounds *still* cause Firefox to crash even today, the other browsers I've tested actually work nicer, more accurate and even faster if I do that, but I really don't want to give up on supporting Firefox.

I'll keep testing and *someday* Firefox will get better, I hope!
